math/lean/Portfile

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+PortGroup           github 1.0
+
+github.setup        leanprover-community lean 3.32.1 v
+revision            0
+
+categories          math
+license             Apache-2
+maintainers         {@kakuhen} openmaintainer
+platforms           darwin
+supported_archs     x86_64 arm64
+description         The Lean theorem prover.
+long_description    A fork of Lean 3 maintained and updated by the Lean \
+    community. Lean is a functional programming language that makes it easy to \
+    write correct and maintainable code. You can also use Lean as an \
+    interactive theorem prover.
+
+depends_lib         port:gmp
+
+checksums \
+    rmd160    6b1b8a1134c40ab772e8a3795ae0b29e13a24a22 \
+    sha256    50c49a426024fa35ee2547f7e82803deecb81fbdc187734da9b0fb56ff54e99f \
+    size      1873303
+
+cmake.build_type    Release
+cmake.generator     Ninja
+
+configure.post_args \
+    ${workpath}/${distname}/src
```

Comment:
I missed this last thing first time through. Although this does work, there is a more "MacPortsy" way to do this, but setting the variable ```${worksrcdir}```. The default ```${worksrcdir}``` is ```${distname}```, so what we do is:
```
${worksrcdir} ${worksrcdir}/src
```

Now i always have to try this to be sure I have it right. Some times you need this construct:
```
set ${worksrcdir} ${worksrcdir}/src
```
and some times you just need this:
```
${worksrcdir} ${worksrcdir}/src
```
and I forget just which one is when.

Then ```${worksrcpath}``` is contructed from ```${workpath}/${worksrcdir}``` and that is passed to cmake by the cmake PortGroup in the post-args.  

i admit, this is confusing for new users, and it might just be easier to do what you did:
```
configure.post_args  ${workpath}/${distname}/src
```
which seems a lot clearer and easier to remember. But that is not MacPorts' way, and we do it this other way in all ports.

Reply:
The reason I used the configure argument is becuase the cmake port group makes a folder called "build," separate from the actual folder that holds the extracted source, and the CMake sources are in a "src" folder inside the extracted folder. I will see what I can do right now to apply your advice.

Reply:
You're 100% right about that. I think you'll find that one of these works. Now that I read it the 3rd time, I think it might be this:
```
worksrcdir ${worksrcdir}/src
```

or perhaps
```
set worksrcdir ${worksrcdir}/src
```

Reply:
The latter option worked on my machine, and the binaries are still functional. I pushed the changes just now.